File: 0031-next-permutation/0031-next-permutation.java
class Solution {
    public List<Integer> helper(int []nums,int i)
    {
        int curr=nums[i];
        int max=0;
        boolean flag=false;
        List<Integer> list=new ArrayList<>();
        int index=-1;
        for(int j=i;j<nums.length;j++)
        {
            if(nums[j]>nums[i])
            {
                if(max>nums[j])
                { max=nums[j];
                index=j-i;
                }
                else if(max==0)
                { max=nums[j];
                 index=j-i;}
            }
            list.add(nums[j]);
        }
        if(index==-1)
            return new ArrayList<>();
        list.remove(index);
        Collections.sort(list);
        nums[i]=max;
        i++;
        for(int j=0;j<list.size();j++)
        {
            nums[i+j]=list.get(j);
        }
        return list;
    }
    public void nextPermutation(int[] nums) {
        int first_max=0,second_max=0;
        int i,j,n=nums.length;
        boolean found=false;
        for(i=n-2;i>=0;i--)
        {
            List<Integer> list=helper(nums,i);
            //System.out.println(list);
            if(list.size()==0)
                continue;
            found=true;
            break;
        }
        if(!found)
            Arrays.sort(nums);
    }
}

File: 0031-next-permutation/0031-next-permutation.py
class Solution:
    def nextPermutation(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        idx = -1
        for i in range(len(nums) - 1, 0, -1):
            if nums[i] > nums[i - 1]:
                idx = i-1
                break
        if idx == -1:
            nums.sort()
        else:
            p = idx
            x=0
            for i in range(idx + 1, len(nums)):
                if nums[idx]<nums[i]: # Corrected condition
                    x = i
            t=nums[idx]
            nums[idx]=nums[x]
            nums[x]=t
            i=idx+1
            j=len(nums)-1
            while i<j:
                t=nums[i]
                nums[i]=nums[j]
                nums[j]=t
                i=i+1
                j=j-1
        return nums